gq/gq_test.go

```diff
@@ -33,10 +34,91 @@ func TestProveVerify(t *testing.T) {
 
 	ok := signerVerifier.VerifyJWT(gqToken)
 	if !ok {
-		t.Fatal("couldn't verify signature we just made")
+		t.Fatal("signature verification failed")
 	}
 }
 
+func TestVerifyModifiedIdPayload(t *testing.T) {
+	oidcPrivKey, err := rsa.GenerateKey(rand.Reader, 2048)
+	if err != nil {
+		t.Fatal(err)
+	}
+
+	oidcPubKey := &oidcPrivKey.PublicKey
+
+	idToken, err := createOIDCToken(oidcPrivKey, "test")
+	if err != nil {
+		t.Fatal(err)
+	}
+
+	// modify the ID Token payload to detect IdP signature invalidity via GQ verify
+	err = modifyTokenPayload(idToken, "fail")
+	if err != nil {
+		t.Fatal(err)
+	}
+
+	signerVerifier := gq.NewSignerVerifier(oidcPubKey, 256)
+	gqToken, err := signerVerifier.SignJWT(idToken)
```

Comment:
Just a thought for another PR - but maybe we should fail here if the `idToken` isn't valid. WDYT?

Reply:
Ah that's not quite what I meant, sorry - I meant maybe we should change `signerVerifier.SignJWT` to error if the idToken doesn't verify with the pubkey. Just a random musing though, not necessary for this PR. The change you made is fine too :smile: